### PR TITLE
recutils: %n crash fix for 10.13

### PR DESCRIPTION
**Pull request by vincecima**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
On the topic of `brew audit --strict`, I am getting an error related to the location of Emacs Lisp files:
```
* Emacs Lisp files were linked directly to /usr/local/share/emacs/site-lisp
    This may cause conflicts with other packages.
    They should instead be installed into:
    /usr/local/opt/recutils/share/emacs/site-lisp/recutils

    The offending files are:
      /usr/local/opt/recutils/share/emacs/site-lisp/rec-mode.el
            /usr/local/opt/recutils/share/emacs/site-lisp/ob-rec.el
```

Not sure how to resolve that one.

### CODE REVIEW

File: Formula/recutils.rb [hunk 1]
```diff
@@ -16,6 +16,13 @@ class Recutils < Formula
     sha256 "aec2464c5e26a561b340e9ae5a080366a068936ff2ba4e86e6a4bcf0ed8a95d0" => :mountain_lion
   end
 
+  if MacOS.version >= :high_sierra
```
Comment (vincecima) [inline Formula/recutils.rb]: `recutils` compiles just fine currently, but most of the `rec*` binaries fail due to how `%n` is used. https://raw.githubusercontent.com/macports/macports-ports/b76d1e48dac/editors/nano/files/secure_snprintf.patch describes the issue and luckily works here as well.

File: Formula/recutils.rb [hunk 2]
```diff
@@ -27,5 +34,13 @@ def install
       1,2,3
     EOS
     system "#{bin}/csv2rec", "test.csv"
+
+    (testpath/"test.rec").write <<~EOS
```
Comment (vincecima) [inline Formula/recutils.rb]: Add this test so future breakages to the main binaries get caught.

**Comment by vincecima** [comment]
On the topic of the `brew audit` failure, looks like we need to pass an additional flag to the `configure` line. Working on a commit for that.

**Comment by vincecima** [comment]
Ok, I am going to need some help with these `lisp` files, bouncing off what the right options are to set between `homebrew` and the pkg's `configure` flags.

**Comment by vincecima** [comment]
Ok, resolved the lisp/emacs issue. Ready for review.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @vincecima! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**